test/unit/JobsManager.js

```diff
@@ -40,176 +47,288 @@ contract("JobsManager", accounts => {
+        })
+    })
+
+    describe("setFinderFee", () => {
+        it("should fail if caller is not Controller owner", async () => {
+            await expectThrow(jobsManager.setFinderFee(10 * PERC_MULTIPLIER, {from: accounts[2]}))
+        })
+
+        it("should fail if provided finderFee is invalid percentage", async () => {
+            await expectThrow(jobsManager.setFinderFee(PERC_DIVISOR + 1))
+        })
+
+        it("should set finderFee", async () => {
+            await jobsManager.setFinderFee(10 * PERC_MULTIPLIER)
+
+            assert.equal(await jobsManager.finderFee.call(), 10 * PERC_MULTIPLIER, "wrong finderFee")
         })
     })
 
     describe("deposit", () => {
         const broadcaster = accounts[0]
 
+        it("should increase broadcaster's deposit", async () => {
+            await jobsManager.deposit({from: broadcaster, value: 1000})
+
+            const bInfo = await jobsManager.broadcasters.call(broadcaster)
+            assert.equal(bInfo[0], 1000, "wrong deposit")
+        })
+    })
+
+    describe("withdraw", () => {
+        const broadcaster = accounts[0]
+        const currentBlock = 100
+        const endBlock = 150
+
         beforeEach(async () => {
-            await jobsManager.setParameters(
-                VERIFICATION_RATE,
-                VERIFICATION_PERIOD,
-                SLASHING_PERIOD,
-                FAILED_VERIFICATION_SLASH_AMOUNT,
-                MISSED_VERIFICATION_SLASH_AMOUNT,
-                DOUBLE_CLAIM_SEGMENT_SLASH_AMOUNT,
-                FINDER_FEE
-            )
-            await fixture.token.setApproved(true)
-        })
-
-        it("should update broadcaster deposit", async () => {
+            await fixture.roundsManager.setMockUint256(functionSig("blockNum()"), currentBlock)
+
             await jobsManager.deposit({from: broadcaster, value: 1000})
-            const bDeposit = (await jobsManager.broadcasters.call(broadcaster))[0]
-            assert.equal(bDeposit, 1000, "broadcaster deposit incorrect")
+            const transcodingOptions = web3.sha3("foo").slice(0, 10) // 0x + first 4 bytes
+            await jobsManager.job("foo", transcodingOptions, 1, endBlock, {from: broadcaster})
+        })
+
+        it("should fail if broadcaster's withdraw block is in the future", async () => {
+            await expectThrow(jobsManager.withdraw({from: broadcaster}))
+        })
+
+        it("should zero out broadcaster's deposit and withdraw block", async () => {
+            // Fast forward through end block
+            await fixture.roundsManager.setMockUint256(functionSig("blockNum()"), endBlock)
+
+            await jobsManager.withdraw({from: broadcaster})
+
+            const bInfo = await jobsManager.broadcasters.call(broadcaster)
+            assert.equal(bInfo[0], 0, "wrong deposit")
+            assert.equal(bInfo[1], 0, "wrong withdraw block")
```

Comment:
The Minter dependency is mocked such that `minter().withdrawETH()` just returns a bool, but the function call itself doesn't do anything because the mocked Minter dependency doesn't actually implement `withdrawETH()` - we just tell the mocked dependency to return a bool when it receives that function call. Testing the ETH transfer triggered by a call to the Minter could be done in an integration test that uses a non-mocked Minter